cupyx/scipy/sparse/linalg/_norm.py

```diff
@@ -0,0 +1,113 @@
+    Returns:
+        ndarray : 0-D or 1-D array or norm(s).
+
+    .. seealso:: :func:`scipy.sparse.linalg.norm`
+    """
+
+    if not cupyx.scipy.sparse.issparse(x):
+        raise TypeError(("input is not sparse. use cupy.linalg.norm"))
+
+    # Check the default case first and handle it immediately.
+    if axis is None and ord in (None, 'fro', 'f'):
+        return _sparse_frobenius_norm(x)
+
+    # Some norms require functions that are not implemented for all types.
+    x = x.tocsr()
+
+    if axis is None:
+        axis = (0, 1)
+    elif not isinstance(axis, tuple):
+        msg = "'axis' must be None, an integer or a tuple of integers"
+        try:
+            int_axis = int(axis)
+        except TypeError:
+            raise TypeError(msg)
+        if axis != int_axis:
+            raise TypeError(msg)
+        axis = (int_axis,)
+
+    nd = 2
+    if len(axis) == 2:
+        row_axis, col_axis = axis
+        if not (-nd <= row_axis < nd and -nd <= col_axis < nd):
+            raise ValueError('Invalid axis %r for an array with shape %r' %
+                             (axis, x.shape))
+        if row_axis % nd == col_axis % nd:
+            raise ValueError('Duplicate axes given.')
+        if ord == 2:
+            raise NotImplementedError
+            # return _multi_svd_norm(x, row_axis, col_axis, amax)
+        elif ord == -2:
+            raise NotImplementedError
+            # return _multi_svd_norm(x, row_axis, col_axis, amin)
+        elif ord == 1:
+            return abs(x).sum(axis=row_axis).max(axis=col_axis).item()
+        elif ord == numpy.Inf:
+            return abs(x).sum(axis=col_axis).max(axis=row_axis).item()
+        elif ord == -1:
+            return abs(x).sum(axis=row_axis).min(axis=col_axis).item()
+        elif ord == -numpy.Inf:
+            return abs(x).sum(axis=col_axis).min(axis=row_axis).item()
+        elif ord in (None, 'f', 'fro'):
+            # The axis order does not matter for this norm.
+            return _sparse_frobenius_norm(x)
+        else:
+            raise ValueError("Invalid norm order for matrices.")
+    elif len(axis) == 1:
+        a, = axis
+        if not (-nd <= a < nd):
+            raise ValueError('Invalid axis %r for an array with shape %r' %
+                             (axis, x.shape))
+        if ord == numpy.Inf:
+            M = abs(x).max(axis=a)
+        elif ord == -numpy.Inf:
+            M = abs(x).min(axis=a)
+        elif ord == 0:
+            # Zero norm
+            M = (x != 0).sum(axis=a)
+        elif ord == 1:
+            # special case for speedup
+            M = abs(x).sum(axis=a)
+        elif ord in (2, None):
+            M = cupy.sqrt(abs(x).power(2).sum(axis=a))
+        else:
+            try:
+                ord + 1
+            except TypeError:
+                raise ValueError('Invalid norm order for vectors.')
+            M = cupy.power(abs(x).power(ord).sum(axis=a), 1 / ord)
+        return M.ravel()
```

Comment:
Could you fix the error `'coo_matrix' object has no attribute 'ravel'`?

Reply:
Maybe the error is caused by the conflict with #3536.  Shall I fix it?

Reply:
how to fix it ?

Reply:
`return M.A.ravel()` (as in [scipy/sparse/linalg/_norm.py#L180](https://github.com/scipy/scipy/blob/b89e2806b0ecd48e128767ab60973fecc69f1f1c/scipy/sparse/linalg/_norm.py#L180)) looks good, while I'm not sure.